modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java

```diff
@@ -475,6 +484,93 @@ public void testCombineSegmentsEntriesOfManualCollectionWithoutDuplications()
 		}
 	}
 
+	@Test
+	public void testGetAssetQueryWithMostRelevantLocale() throws Exception {
+		Group group = GroupTestUtil.addGroup();
+
+		GroupTestUtil.updateDisplaySettings(
+			group.getGroupId(), Arrays.asList(LocaleUtil.US, LocaleUtil.SPAIN),
+			LocaleUtil.US);
+
+		DataDefinition dataDefinition =
+			DataDefinitionTestUtil.addDataDefinition(
+				"journal", _dataDefinitionResourceFactory, group.getGroupId(),
+				_read("data_definition.json"), TestPropsValues.getUser());
+
+		JournalTestUtil.addArticleWithXMLContent(
+			group.getGroupId(), JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			JournalArticleConstants.CLASS_NAME_ID_DEFAULT,
+			_readFileToString("journal_article_content.xml"),
+			dataDefinition.getDataDefinitionKey(), null, LocaleUtil.US);
+
+		Locale originalThemeDisplayLocale =
```

Comment:
in the try block it would be a good idea to set the threadLocal to US, this in case there is some leftover from other test cases. The test might become flaky if we are not careful